dhcpv4/nclient4/client.go

```diff
@@ -132,9 +125,12 @@ func NewWithConn(conn net.PacketConn, ifaceHWAddr net.HardwareAddr, opts ...Clie
 		opt(c)
 	}
 
+	if c.conn == nil {
```

Comment:
nit: could check early if `conn != nil` at the beginning of the function instead?